songscope_web/src/pages/api/spotify-test.js

```diff
@@ -0,0 +1,43 @@
+// Test route for plumbing project, delete later
+require("dotenv").config();
+
+const client_id = process.env.SPOTIFY_CLIENT_ID;
+const client_secret = process.env.SPOTIFY_CLIENT_SECRET;
+const auth_token = Buffer.from(`${client_id}:${client_secret}`).toString("base64");
+
+function toUrlEncoded(obj) {
+    return Object.keys(obj)
+        .map((key) => encodeURIComponent(key) + "=" + encodeURIComponent(obj[key]))
+        .join("&");
+}
+
+const getAuth = async () => {
+    const response = await fetch("https://accounts.spotify.com/api/token", {
+        method: "POST",
+        headers: {
+            "Authorization": `Basic ${auth_token}`,
+            "Content-Type": "application/x-www-form-urlencoded"
+        },
+        body: toUrlEncoded({grant_type: "client_credentials"})
+    });
+    return JSON.parse(await response.text()).access_token;
```

Comment:
some issue with the .json() parser not working so I just put it as 
```js
JSON.parse(await response.text())
```
instead, other than that everything looks good, merging now